Reword "The humans behind Kedro" section of README (#2343)
kedro-org/kedro: README.md

## The humans behind Kedro

> Kedro is maintained by a [product team](https://kedro.readthedocs.io/en/stable/faq/faq.html) and a number of [contributors from across the world](https://github.com/kedro-org/kedro/releases).

Reviewer: ```suggestion
[The Kedro product team](https://kedro.readthedocs.io/en/stable/faq/faq.html#who-maintains-kedro) and a number of [open source contributors from across the world](https://github.com/kedro-org/kedro/releases) maintain Kedro.
```

Reviewer: While we're changing the link, let's make it even more accessible and add a bit more context to the link description.
